advanced/Templates/pihole-FTL.service.ini

```
@@ -0,0 +1,37 @@
+[Unit]
+Description=pihole-FTL Domain Name Server
+Documentation=man:pihole-FTL(8)
+After=remote-fs.target
+
+# Alternative based on https://wiki.debian.org/Hardening/Daemons#bind
+#   allows other services which rely on DNS to use pihole-FTL, but means
+#   breaking if config relies on remote-fs
+# After=network.target
+# Wants=nss-lookup.target
+# Before=nss-lookup.target
+
+After=resolvconf.service
+Wants=resolvconf.service
+# fail since dnsmasq is embedded
+Conflicts=dnsmasq.service
+
+[Service]
+# Run as unprivileged user, use AmbientCapabilities to assign capabilities
+User=pihole
+Group=pihole
+
+# start()
+ExecStartPre=+/bin/sh /etc/pihole/fix_files.sh
+ExecStartPre=+/bin/sh -c "echo \"nameserver 127.0.0.1\" | /sbin/resolvconf -a lo.piholeFTL"
+# Assign capabilities to the pihole user slice
+AmbientCapabilities=CAP_NET_BIND_SERVICE CAP_NET_RAW CAP_NET_ADMIN
+ExecStart=/usr/bin/pihole-FTL no-daemon
+
+## stop()
+# Clean up resolv.conf before shutting down service
+ExecStop=+/sbin/resolvconf -d lo.piholeFTL
+# Send SIGKILL 5 seconds after SIGTERM
+TimeoutStopSec=5
```

We might want to give FTL a bit more time. After receiving `SIGTERM`, it may still have to write a lot of history to the database. In a previous PR, I proposed a limit of 10 seconds here.
```suggestion
TimeoutStopSec=10
```

---

It would probably be better as a separate PR since you'd want to change the init script at the same time?

---

Yes, we do this separately.